agent/api/statechange.go

```diff
@@ -72,6 +73,69 @@ type TaskStateChange struct {
 	Task *Task
 }
 
+// NewTaskStateChangeEvent creates a new task state change event
+func NewTaskStateChangeEvent(task *Task, reason string) (TaskStateChange, error) {
+	var event TaskStateChange
+	taskKnownStatus := task.GetKnownStatus()
+	if !taskKnownStatus.BackendRecognized() {
+		return event, errors.Errorf(
+			"create task state change event api: status not recognized by ECS: %v; task: %s",
+			taskKnownStatus, task.Arn)
```

Comment:
We can avoid having the `task.Arn` here and in below cases since we are already logging them in catch.

Reply:
right, thanks